package/kernel/linux/modules/lib.mk

```diff
@@ -98,6 +98,132 @@ endef
+
+define KernelPackage/lib-crypto-chacha20/arm64
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_NEON
+  FILES+=$(LINUX_DIR)/arch/arm64/crypto/chacha-neon.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/mips
+  KCONFIG+=CONFIG_CRYPTO_CHACHA_MIPS
+  FILES:=$(LINUX_DIR)/arch/mips/crypto/chacha-mips.ko
+endef
+
+KernelPackage/lib-crypto-chacha20/ath79=$(KernelPackage/lib-crypto-chacha20/mips)
+KernelPackage/lib-crypto-chacha20/ramips=$(KernelPackage/lib-crypto-chacha20/mips)
+
+$(eval $(call KernelPackage,lib-crypto-chacha20))
+
+
+define KernelPackage/lib-crypto-chacha20poly1305
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha20-Poly1305 AEAD support (8-byte nonce library version)
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA20POLY1305
+  DEPENDS:=+kmod-lib-crypto-chacha20 +kmod-lib-crypto-poly1305
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha20poly1305.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-chacha20poly1305))
+
+
+define KernelPackage/lib-crypto-curve25519
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=Curve25519 scalar multiplication library
+  KCONFIG:=CONFIG_CRYPTO_LIB_CURVE25519
+  FILES:= \
+	$(LINUX_DIR)/lib/crypto/libcurve25519.ko \
+	$(LINUX_DIR)/lib/crypto/libcurve25519-generic.ko
+endef
+
+define KernelPackage/lib-crypto-curve25519/x86/64
+  KCONFIG+=CONFIG_CRYPTO_CURVE25519_X86
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/curve25519-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-curve25519/arm-neon
+  KCONFIG+=CONFIG_CRYPTO_CURVE25519_NEON
+  FILES+=$(LINUX_DIR)/arch/arm/crypto/curve25519-neon.ko
+endef
+
+KernelPackage/lib-crypto-curve25519/imx6=$(KernelPackage/lib-crypto-curve25519/arm-neon)
+KernelPackage/lib-crypto-curve25519/ipq40xx=$(KernelPackage/lib-crypto-curve25519/arm-neon)
+KernelPackage/lib-crypto-curve25519/mvebu/cortexa9=$(KernelPackage/lib-crypto-curve25519/arm-neon)
+
+$(eval $(call KernelPackage,lib-crypto-curve25519))
+
+
+define KernelPackage/lib-crypto-poly1305
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=Poly1305 library interface
+  KCONFIG:=CONFIG_CRYPTO_LIB_POLY1305
+  FILES:=$(LINUX_DIR)/lib/crypto/libpoly1305.ko
+endef
+
+define KernelPackage/lib-crypto-poly1305/x86/64
+  KCONFIG+=CONFIG_CRYPTO_POLY1305_X86_64
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/poly1305-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-poly1305/arm
+  KCONFIG+=CONFIG_CRYPTO_POLY1305_ARM
+  FILES:=$(LINUX_DIR)/arch/arm/crypto/poly1305-arm.ko
+endef
+
+KernelPackage/lib-crypto-poly1305/kirkwood=$(KernelPackage/lib-crypto-poly1305/arm)
+
+define KernelPackage/lib-crypto-poly1305/arm64-neon
+  KCONFIG+=CONFIG_CRYPTO_POLY1305_NEON
+  FILES:=$(LINUX_DIR)/arch/arm64/crypto/poly1305-neon.ko
+endef
+
+define KernelPackage/lib-crypto-poly1305/mips
+  KCONFIG+=CONFIG_CRYPTO_POLY1305_MIPS
+  FILES:=$(LINUX_DIR)/arch/mips/crypto/poly1305-mips.ko
+endef
+
+KernelPackage/lib-crypto-poly1305/ath79=$(KernelPackage/lib-crypto-poly1305/mips)
+KernelPackage/lib-crypto-poly1305/ramips=$(KernelPackage/lib-crypto-poly1305/mips)
```

Comment:
The poly code can include all the other 32bit mips platforms PLUS the 64bit mips platforms, like `/octeon`.